File: velen-saas-authority/velen-saas-authority-biz/src/main/java/com/github/binmagic/saas/velen/authority/service/AppService.java
package com.github.binmagic.saas.velen.authority.service;

import com.github.binmagic.saas.velen.authority.dto.AppAddMemberDTO;
import com.github.binmagic.saas.velen.authority.dto.AppMemberInfoDTO;
import com.github.binmagic.saas.velen.authority.entity.App;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

public interface AppService
{
	Flux<App> findApp(String account);

	Mono<App> createApp(App app);

	Mono<Void> submitToAuditor(String appId);

	Mono<App> updateApp(App app);

	Mono<App> getApp(String appId);

	Flux<AppMemberInfoDTO> getAppMember(String appId);

	Mono<String> getAppMemberRole(String appId, String account);

	Mono<Void> addMember(String appId, List<AppAddMemberDTO> appAddMemberDTO);

	Mono<Void> deleteMember(String appId, String id);
}

File: velen-saas-authority/velen-saas-authority-biz/src/main/java/com/github/binmagic/saas/velen/authority/service/AppService.kt
package com.github.binmagic.saas.velen.authority.service

import com.github.binmagic.saas.velen.authority.dto.AppAddMemberDTO
import com.github.binmagic.saas.velen.authority.dto.AppMemberInfoDTO
import com.github.binmagic.saas.velen.authority.entity.App
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface AppService {

    suspend fun findApp(account : String) :Flux<App>

    suspend fun createApp(app:App) :Mono<App>

    suspend fun submitToAuditor(appId : String) : Mono<Void>

    suspend fun updateApp(app:App) : Mono<App>

    suspend fun getApp(appId: String) : Mono<App>

    suspend fun getAppMember(appId: String) : Flux<AppMemberInfoDTO>

    suspend fun getAppMemberRole(appId: String,account: String) : Mono<String>

    suspend fun addMember(appId: String, appAddMemberDTO: List<AppAddMemberDTO>) : Mono<Void>

    suspend fun deleteMember(appId: String,id:String) : Mono<Void>

    suspend fun deleteApp(appId: String) : Mono<Void>
}